README.md

```diff
@@ -142,46 +152,60 @@ There is no strict limit over what you can do with a `Matrix`,
+the specified value. Supported values are most of the Android `Gravity` flags, plus `TRANSFORMATION_GRAVITY_AUTO`.
 
 |Transformation Gravity|Description|
 |----------------------|-----------|
-|`top`|If the content is taller than the view, translate it so that we see the top part.|
-|`bottom`|If the content is taller than the view, translate it so that we see the bottom part.|
-|`left`|If the content is wider than the view, translate it so that we see the left part.|
-|`right`|If the content is wider than the view, translate it so that we see the right part.|
+|`top`, ...|The content is panned so that its *top* side matches teh container *top* side. Same for other values.|
+|`auto` (default)|The transformation gravity is taken from the engine [alignment](#alignment), defaults to `center` on both axes.|
 
-If, after this process, the content is smaller than the container, note that the current
-[Smaller Policy](#smaller-policy) applies.
+**Note: after transformation and gravity are applied, the engine will apply - as always - all the active constraints,
+including minZoom, maxZoom, alignment. This means that the final position might be slightly (or completely) different.**
 
-Note: you can always trigger a new transformation to be applied by using the `setContentSize` or `setContainerSize` APIs.
+For example, when `maxZoom == 1`, is forced to not be any larger than the container. This means that
+a `centerCrop` transformation will not work (will act just like a `centerInside`).
 
-#### Smaller policy
+#### Alignment
 
-You can control how the content will be positioned when it is smaller than the container through 
-the `setSmallerPolicy(int)` method or by using the `smallerPolicy` XML attribute of `ZoomLayout` and `ZoomImageView`.
-By default, content is always centered when it is smaller than its container.
+You can force the content position with respect to the container using the `setAlignment(int)` method
+or the `alignment` XML flag of `ZoomLayout` and `ZoomImageView`. 
+The default value is `Alignment.CENTER` which will center the content on both directions.
 
-|Policy|Description|
-|------|-----------|
-|`center`|The content will be centered within the container.|
-|`fromTransformation`|The content will respect the gravity parameter of the transformation (which defaults to `Gravity.CENTER` as well).|
-|`none`|The content is free to be moved around inside the container bounds.|
+**Note: alignment does not make sense when content is larger than the container, because forcing an 
+alignment (e.g. left) would mean making part of the content unreachable (e.g. the right part).**
+
+|Alignment|Description|
+|---------|-----------|
+|`top`, `bottom`, `left`, `right`|Force align the content to the same side of the container.|
+|`center_horizontal`, `center_vertical`|Force the content to be centered inside the container on that axis.|
+|`none_horizontal`, `none_vertical`|No alignment set: content is free to be moved on that axis.|
```

Comment:
Are you going with those constants nonetheless?
Ah so they will be used internally but the dev user doesn't have to specify them. :+1:

Reply:
Yes, we need them internally so there's no point in hiding them. However user doesn't have to specify explicitly